rp2040-hal/src/spi.rs

```diff
@@ -165,14 +200,15 @@ impl<D: SpiDevice, const DS: u8> Spi<Disabled, D, DS> {
     }
 
     /// Set format and datasize
-    fn set_format(&mut self, data_bits: u8, mode: &Mode) {
+    fn set_format<M: Into<Mode> + Clone>(&mut self, data_bits: u8, mode: &M) {
+        let mode: Mode = (*mode).clone().into();
```

Comment:
It would be a breaking change but 0.9.0 is planned to have loads anyway, so this can be made into:
```rust
    fn set_format<M: Into<Mode>>(&mut self, data_bits: u8, mode: M) {
        let mode: Mode = mode.into();
```